the-collab-lab/tcl-18-smart-shopping-list
# Va ma purchase status of items

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |

## Description

As a user, I want to view a list of my shopping list items in order of how soon I am likely to need to buy each of them again so that it’s clear what I need to buy soon. 

Possible item states are as follows:

- Need to buy soon (fewer than 7 days)
- Need to buy kind of soon (between 7 & 30 days)
- Need to buy not soon (more than 30 days)
- Inactive (when there’s only 1 purchase in the database or the purchase is really out of date [the time that has elapsed since the last purchase is 2x what was estimated])

## Related Issue

Fix #12 

## Acceptance Criteria


## Changes 
-  There is a new field on the database named creationDate:
![InkedImage+2021-02-12+at+5 14 49+PM_LI](https://user-images.githubusercontent.com/60724393/107827998-e4034780-6d55-11eb-8735-355026949c6f.jpg)
- Some styles were added so that the links do not overlap with the list and more, if the list becomes long, this makes it easier for the screen reader to read the items and their statuses:
![links-lista](https://user-images.githubusercontent.com/60724393/107719117-7ac3fb80-6ca5-11eb-9857-9107ffe47cf7.jpg)


## Testing Steps / QA Criteria

1. Add a new item (if you click on the radio buttons that are there, either Soon, Kind of soon or Not soon, they will be displayed in the list with a different color: Green, orange and red respectively, this according to how soon we should buy the item).
2. In the case of inactive, remember that the time that has passed since the last purchase is 2 times the estimated number when the purchase was due. Something to keep in mind, there is a new field on the database named creationDate, if an item that have not been purchased it's too old, it will also got inactive. Try this:

- Add a product "soon"
- Give it check
- Then go to te database and change the creation date to 01/Feb/2021
- Check the item list, you'll see that this product is inactivated

<img width="430" alt="Image+2021-02-11+at+9 01 32+PM" src="https://user-images.githubusercontent.com/60724393/107722759-a64ae400-6cad-11eb-9723-ea9b1b032aa4.png">

3. **Note:** You will see that even if you check it, it will remain with the color corresponding to the purchase status.

### Before
![before](https://user-images.githubusercontent.com/60724393/107719453-4735a100-6ca6-11eb-87ef-58b099dedb6e.png)


### After
As you can see in the video the colors are ordered with respect to when you need to make the purchase, also alphabetically and the screen reader reads and depending on the above says for example: "You need to buy this Soon, dessert":

https://user-images.githubusercontent.com/60724393/107721239-28d1a480-6caa-11eb-836f-5dc7d20c9621.mp4

https://user-images.githubusercontent.com/60724393/107793887-ccf83180-6d24-11eb-9cf4-91ca7e2cba28.mp4
